### PR TITLE
Pool can have a companion ticker thread

### PR DESCRIPTION
**Pull request by matthieun**
### Description:

This change brings the ability to schedule a companion ticker thread that allows time-based reporting on the initial Runnable or Callable execution.

### Potential Impact:

No impact, it is a new feature.

### Unit Test Approach:

Tested the new option with Callable and Runnable, and made sure they still report errors properly.

### Test Results:

Companion thread has the same lifespan as other thread.

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/utilities/threads/Pool.java [hunk 1]
```diff
@@ -126,11 +126,45 @@ public boolean isDead()
...
+            try
+            {
+                command.run();
+            }
+            finally
+            {
+                ticker.close();
+            }
+        };
+        this.pool.execute(ticker);
```
Comment (lucaspcram) [inline src/main/java/org/openstreetmap/atlas/utilities/threads/Pool.java]: One thing to note here. When using the `Pool` with a `Runnable`, it fails fast if the `Runnable` throws an exception. So you could have a case where the `ticker` throws an exception before the `queue` method even gets to the next line.

Comment (matthieun) [reply src/main/java/org/openstreetmap/atlas/utilities/threads/Pool.java]: Actually you are right, let me add ticker error handling!

**Comment by matthieun** [comment]
Rhaaaaaa!

File: src/test/java/org/openstreetmap/atlas/utilities/threads/PoolTest.java [hunk 2]
```diff
@@ -138,6 +138,48 @@ public void testTickerCallableWithErrors()
...
+                    final Ticker rogueTicker = new Ticker("Ticker " + idx,
+                            Duration.milliseconds(50))
+                    {
+                        @Override
+                        protected void tickAction(final Duration sinceStart)
+                        {
+                            logger.info("{}: {}", getName(), sinceStart);
+                            if ("Ticker 3".equals(getName()))
+                            {
+                                throw new CoreException("I am rogue Ticker 3");
```
Comment (lucaspcram) [inline src/test/java/org/openstreetmap/atlas/utilities/threads/PoolTest.java]: Hahaha "rogue ticker" I love this :D

**Pull request merged**